### PR TITLE
dotCMS/core#19088 Update the template designer for create and edit

### PR DESCRIPTION
**Pull request by fmontes**
Implementing state manager for the component and adding everything need it.

### CODE REVIEW

File: src/app/api/services/dot-router/dot-router.service.ts [hunk 1]
```diff
@@ -107,6 +107,16 @@ export class DotRouterService {
         this.router.navigate([`/${portlet}/edit/${id}`]);
     }
 
+    /**
+     * Redirecto to edit the template id passed
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-router/dot-router.service.ts:111]: typo

File: src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts [hunk 2]
```diff
@@ -251,58 +184,64 @@ export class DotEditLayoutDesignerComponent implements OnInit, OnDestroy {
...
+                },
+                (error) => this.errorHandler(error)
+            );
+
+        this.initialFormValue = _.cloneDeep(this.form.value);
+        this.isModelUpdated = false;
     }
 
-    // tslint:disable-next-line:cyclomatic-complexity
+    // tslint:disable-next-line: cyclomatic-complexity
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-edit-page/layout/dot-edit-layout-designer/dot-edit-layout-designer.component.ts:236]: is this needed?

File: src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.spec.ts [hunk 3]
```diff
@@ -11,77 +11,65 @@ import { TestBed } from '@angular/core/testing';
 describe('DotEditLayoutService', () => {
     const containers = {
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.spec.ts:12]: can you put all containers data into a single file to be reused?
src/app/portlets/dot-edit-page/template-containers-cache.spec.ts

File: src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                {
+                    label: 'Save',
+                    disabled: disabled,
+                    command: () => {
+                        this.store.saveTemplate(this.form.value);
+                    }
+                }
+            ],
+            cancel: () => {
+                console.log('cancel');
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts:134]: remove

File: src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+                this.setState({
+                    original: template,
+                    working: template,
+                    type: fixType
+                });
+            });
+    }
+
+    cancelCreate = () => {
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts:184]: doc

File: src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts [hunk 6]
```diff
@@ -39,8 +65,8 @@ export class DotTemplateListComponent implements OnInit, OnDestroy {
      *
      * @memberof DotTemplateListComponent
      */
-    editTemplate(template: DotTemplate): void {
-        console.log(template);
+    editTemplate({ identifier }: DotTemplate): void {
```
Comment (alfredo-dotcms) [inline src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts:68]: update doc

File: src/app/view/components/dot-form-dialog/dot-form-dialog.component.html [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <div class="dot-form-dialog__content">
-    <ng-content></ng-content>
+    <ng-content pFocusTrap></ng-content>
```
Comment (alfredo-dotcms) [inline src/app/view/components/dot-form-dialog/dot-form-dialog.component.html:2]: what is `pFocusTrap ` for?

File: src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts (hunk 4 above)
Comment (fmontes) [reply src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts:134]: not yet.

File: src/app/view/components/dot-form-dialog/dot-form-dialog.component.html (hunk 7 above)
Comment (fmontes) [reply src/app/view/components/dot-form-dialog/dot-form-dialog.component.html:2]: Is an primeng directive that will trap the focus of the elements inside, in this case we have a form in the dialog and I want that when the user hit `tab` it only focus the fields of this form and not the ones outside.

File: src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts [hunk 8]
```diff
@@ -4,22 +4,48 @@ import { pluck, takeUntil } from 'rxjs/operators';
...
 
     ngOnInit(): void {
+        this.actions = {
+            primary: {
+                model: [
+                    {
+                        command: () => {
+                            this.dotRouterService.gotoPortlet('/templates/new/designer');
+                        },
+                        label: 'Designer'
```
Comment (hmoreras) [inline src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts:37]: this.dotMessageService.get('design-template')

File: src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts [hunk 9]
```diff
@@ -4,22 +4,48 @@ import { pluck, takeUntil } from 'rxjs/operators';
...
+                    },
+                    {
+                        command: () => {
+                            this.dotRouterService.gotoPortlet('/templates/new/advanced');
+                        },
+                        label: 'Advanced'
```
Comment (hmoreras) [inline src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts:43]: this.dotMessageService.get('code-template')

File: src/app/view/components/dot-edit-layout-designer/dot-edit-layout-designer.component.ts [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+                    this.cd.detectChanges();
+                },
+                (error) => this.errorHandler(error)
+            );
+
+        this.initialFormValue = _.cloneDeep(this.form.value);
+        this.isModelUpdated = false;
+    }
+
+    // tslint:disable-next-line: cyclomatic-complexity
```
Comment (hmoreras) [inline src/app/view/components/dot-edit-layout-designer/dot-edit-layout-designer.component.ts:240]: remove ?

File: src/app/view/components/dot-edit-layout-designer/dot-edit-layout-designer.module.ts [hunk 11]
```diff
@@ -1,31 +1,37 @@
-import { DotIconButtonModule } from './../../../../view/components/_common/dot-icon-button/dot-icon-button.module';
+import { DotIconButtonModule } from '../_common/dot-icon-button/dot-icon-button.module';
```
Comment (hmoreras) [inline src/app/view/components/dot-edit-layout-designer/dot-edit-layout-designer.module.ts:1]: @components in few includes in this file

**Review by hmoreras — CHANGES_REQUESTED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**